### PR TITLE
Add initial tracking capability

### PR DESCRIPTION
**Pull request by aeirola**
Integrates with `matomo-sdk-ios` and `matomo-sdk-android` libraries, and provides screen and event tracking. More features may be added in later PRs.

### CODE REVIEW

File: android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java [hunk 1]
```diff
@@ -19,4 +32,57 @@ public RNMatomoSdkModule(ReactApplicationContext reactContext) {
...
+      if (tracker == null) {
+        tracker = TrackerBuilder
+                .createDefault(apiUrl, siteId)
+                .build(Matomo.getInstance(this.reactContext));
+
+        // Disable default automatic user ID generation for consistency with iOS
+        tracker.setUserId(null);
+      }
+      promise.resolve(null);
+    } catch(Exception e) {
```
Comment (TuukkaP) [inline android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java:49]: Do we want to actually catch all errors?

File: android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java [hunk 2]
```diff
@@ -19,4 +32,57 @@ public RNMatomoSdkModule(ReactApplicationContext reactContext) {
...
+      promise.reject(e);
+    }
+  }
+
+  @ReactMethod
+  @SuppressWarnings("unused")
+  public void trackView(ReadableArray route, Promise promise) {
+    try {
+      List<String> routeList = new LinkedList<>();
```
Comment (TuukkaP) [inline android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java:58]: I think `ArrayList<>` would suffice. Not that it matters that much...

Comment (TuukkaP) [reply android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java:58]: https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableArray.java#L31 ?

File: android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java [hunk 3]
```diff
@@ -19,4 +32,57 @@ public RNMatomoSdkModule(ReactApplicationContext reactContext) {
...
+      if (optionalParameters.hasKey("name")) {
+        event.name(optionalParameters.getString("name"));
+      }
+      if (optionalParameters.hasKey("value")) {
+        event.value((float) optionalParameters.getDouble("value"));
+      }
+      event.with(tracker);
+
+      promise.resolve(null);
+    } catch(Exception e) {
```
Comment (TuukkaP) [inline android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java:84]: Catch all?

**Review by TuukkaP — APPROVED** [review]
Looks good to me! 👍  Some minor comments. 

Haven't done much regarding swift but it has similar structure as it's java counterpart thus error handling could be more exact.

File: android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java (hunk 1 above)
Comment (aeirola) [reply android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java:49]: Yeah, I think it is better to propagate all errors to the JS runtime, and handle them there instead.

File: android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java (hunk 2 above)
Comment (aeirola) [reply android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java:58]: Hmm, that would be handy. For some reason gradle grabbed some rather old version of react-native for the dependency. Using newer versions of react in development would probably be easier after https://github.com/terveystalo/react-native-matomo-sdk/pull/2. So maybe I'll fix this later instead.

**Comment by pheis** [comment]
Looks good to me!

**Pull request merged**